### PR TITLE
Upgrade deps (including Go 1.20)

### PR DESCRIPTION
**Pull request by maelvls**
Before cutting the version v0.2.0, I'd like to upgrade the dependencies and the Go version. 

Closes #18, #19, #20.

### CODE REVIEW

**Comment by maelvls** [comment]
@irbekrm Can you take a look? Thanks!

File: .github/workflows/goreleaser.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v3
         with:
-          go-version: "^1.18"
+          go-version: "^1.20"
```
Comment (inteon) [inline .github/workflows/goreleaser.yaml:27]:
```suggestion
          go-version-file: go.mod
          cache: true
```
Can we use this config instead (similar to our other projects)?

**Review by inteon — CHANGES_REQUESTED** [review]

File: .github/workflows/goreleaser.yaml (hunk 1 above)
Comment (inteon) [reply .github/workflows/goreleaser.yaml:27]: + we then probably want to increase the go version in go.mod to 1.20

Comment (maelvls) [reply .github/workflows/goreleaser.yaml:27]: Why should we force anyone else building openshift-routes to use Go 1.20 by setting `go 1.20` in go.mod? The go.mod `go 1.18` indicates a compatibility, and the earliest Go version that can build openshift-routes is 1.18, so I wouldn't set 1.20 here

Comment (maelvls) [reply .github/workflows/goreleaser.yaml:27]: Regarding `cache: true`, you are correct. Alternatively, we could use `actions/setup-go@v4` which enables caching by default

Comment (inteon) [reply .github/workflows/goreleaser.yaml:27]: ok, thx; I approved the PR already.

**Pull request merged**